api/v1alpha2/linodecluster_types.go

```diff
@@ -47,6 +47,11 @@ type LinodeClusterSpec struct {
 	// +optional
 	VPCRef *corev1.ObjectReference `json:"vpcRef,omitempty"`
 
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
```

Comment:
I wonder if we should make it immutable or it can be changed later on. I am ok with either approach, just want to know what others think about it.

Reply:
I think either is fine. Not making it immutable makes it more flexible but not sure what use cases that would be useful in